app/src/main/java/com/sdpteam/connectout/utils/Result.java

```diff
@@ -0,0 +1,43 @@
+package com.sdpteam.connectout.utils;
+
+/**
+ * It contains the value of the operation if it is successful!
+ * This is a immutable object that describe the result of an operation (that return some value of any type)
+ * (is it successful or not, can have an optional message (for instance can be the message error)
+ *
+ * @param <T> type of the value
+ */
+public class Result<T> {
+    private final T value;
+    private final boolean isSuccess;
+    private final String msg;
+
+    public Result(T value, boolean isSuccess, String msg) {
+        if (value == null && isSuccess) {
+            throw new IllegalArgumentException("value cannot be null if isSuccess is true");
+        }
+        if (msg == null) {
+            throw new IllegalArgumentException("msg cannot be null");
+        }
+
+        this.value = value;
+        this.isSuccess = isSuccess;
+        this.msg = msg;
+    }
+
+    public Result(T value, boolean isSuccess) {
+        this(value, isSuccess, "");
+    }
+
+    public T value() {
+        return value;
+    }
+
+    public boolean isSuccess() {
+        return isSuccess;
+    }
+
+    public String msg() {
+        return msg;
+    }
```

Comment:
I like the idea behind this class, however it leaves me perplex too, 
It is clearly useful to know state of fetch, wether it failed or was successful.
However, it might over increase code complexity on both model and viewModel.
The view model in other classes such as `EventViewModel` or `ProfileListViewModel` already takes care of an unsuccessful fetching by using a Null container (Null profile, empty list ....).

I would therefore argue to not use it, but would really like to know your opinion on this class.

Reply:
With this model, it can still produce empty lists (empty list is not a failure!).
Here a failed Result would mean that there was a network issue or some other fatal error that firebase library can return us (it's called addOnFailureListener()). Until now we never handled errors of firebase, but we should. 

This class in our model would describe the two kind of results : errors or success value. Then yes in the viewmodel when you get a Result you should simply check if it `isSuccess()` before taking the value with `.value()` and feeding it into the MutableLiveData. 
And a fun thing we could have in our view model is it it is not a success just trigger a toast with the error message `.msg()`. for example by having a `MutableLiveData<String> errorMsg` that from our view we could listen to 
```
viewModel.getErrorMsg().observeForever(s -> {
            if (s.length() > 0) {
                Toast.makeText(getActivity(), s, Toast.LENGTH_SHORT).show();
            }
```

But of course showing this error message is not mandatory but at least the model provides the possibility and it is the way to go.

Reply:
(Fun fact, I inspired myself from how they do that in Scala, if you recall we saw this last year while working with Akka :
https://www.baeldung.com/scala/future-success-failure#2-recovering-with-an-asynchronous-computation)

Reply:
I see your point, thank you for your detailed explanation !